src/view-strategy.js

```diff
@@ -253,3 +254,77 @@ export class InlineViewStrategy {
     return viewEngine.loadViewFactory(entry, compileInstruction, loadContext, target);
   }
 }
+
+@viewStrategy()
+export class StaticViewStrategy {
+
+  template: string | HTMLTemplateElement;
+  dependencies: Function[] | { (): Function[] | (Function | Promise<Function[] | Record<string, Function>>)[] };
+  factoryIsReady: boolean;
+  factory: ViewFactory;
+  onReady: Promsie<any>;
+
+  constructor(config: string | HTMLTemplateElement | { template: string; dependencies?: any } ) {
+    if (typeof config === 'string' || config instanceof HTMLTemplateElement) {
+      config = {
+        template: config
+      };
+    }
+    this.template = config.template;
+    this.dependencies = config.dependencies || [];
+    this.factoryIsReady = false;
+    this.onReady = null;
+  }
+
+  /**
+  * Loads a view factory.
+  * @param viewEngine The view engine to use during the load process.
+  * @param compileInstruction Additional instructions to use during compilation of the view.
+  * @param loadContext The loading context used for loading all resources and dependencies.
+  * @param target A class from which to extract metadata of additional resources to load.
+  * @return A promise for the view factory that is produced by this strategy.
+  */
+  loadViewFactory(viewEngine: ViewEngine, compileInstruction: IViewCompileInstruction, loadContext: IResourceLoadContext, target: any): Promise<IViewFactory> {
+    if (this.factoryIsReady) {
+      return Promise.resolve(this.factory);
+    }
+    let deps = this.dependencies;
+    deps = typeof deps === 'function' ? deps() : (deps ? Array.isArray(deps) ? deps : [deps] : []);
+    this.onReady = Promise.all(deps).then((dependencies) => {
+      const container = viewEngine.container;
+      const appResources = viewEngine.appResources;
+      const viewCompiler = viewEngine.viewCompiler;
+      const viewResources = new ViewResources(appResources);
+
+      let resource;
+      let elDeps = [];
+
+      if (target) {
+        // when composing without a view mode, but view specified, target will be undefined
+        viewResources.autoRegister(container, target);
+      }
+
+      for (let dep of dependencies) {
+        if (typeof dep === 'function') {
+          resource = viewResources.autoRegister(container, dep);
+        } else if (dep && typeof dep === 'object') {
+          for (let key in dep) {
+            resource = viewResources.autoRegister(container, dep[key]);
```

Comment:
So... I don't quite get it.
What's the point in supporting a record if we do nothing with `key`?
Is it so that future versions could lazy-load resources when their name is first encountered in a view?

That's a cool future feature, but then I think the best choice is to not accept a record until then. 
Because lazy loading is a breaking change, so we don't want people to use this syntax until we actually implement the feature.

Even so, it's not implemented correctly. As lazy-loading implies a factory, a correct eager-loading implementation would be: `dep[key]()`.

Reply:
Oh wait, I think I get it now... 
This is for the case `() => import("y")`, isn't it?
The record is actually an ES module?

Makes sense then. 
I like how we try to find all resources in said module, rather than just the first one as in older Aurelia code 👍 
No reliance on `default`, then.

Reply:
Yes, It is also consistent with the way we currently load resources if it was string: every thing will be registered. And sure, docs for this work gonna be fun...